mlflow/utils/credentials.py

```diff
@@ -76,29 +77,25 @@ def login(backend="databricks"):
         )
 
 
-def _check_databricks_auth():
-    # Check if databricks credentials are set.
-    try:
-        from databricks.sdk import WorkspaceClient
-    except ImportError:
-        raise ImportError(
-            "Databricks SDK is not installed. To use `mlflow.login()`, please install "
-            "databricks-sdk by `pip install databricks-sdk`."
-        )
+def _validate_databricks_auth():
+    """Validate Databricks authentication."""
+    max_trials = 3  # Allow 3 retrys for timeout.
+    timeout = 3
+    for _ in range(max_trials):
+        try:
+            result = subprocess.run(
+                ["databricks", "tokens", "list"],
+                timeout=timeout,
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
```

Comment:
stdout and stderr are useful for debugging issues. Can we include it in the error message?

Reply:
I was keeping stdout/stderr piped initially, but it makes the logging super messy. We are calling the validation function twice, and the first try is to validate existing auth, which shouldn't provide much logging if users don't have anything set up before. 

stdout is not useful here, and stderr is just "your auth is invalid", which does not give much information.

Reply:
> it makes the logging super messy

IMO, that's better than no information on what went wrong.

Reply:
Can we show stdout and stderr in the final attempt?

Reply:
we probably don't want to show users the output of `databricks token list`, which is for checking auth. 

What we are logging is the superset of the error msg, databricks CLI's error is just this:
```
Error: Authorization failed. Your token may be expired or lack the valid scope
```

Reply:
Is `Error: Authorization failed. Your token may be expired or lack the valid scope` the only error `databricks token list` throws?

Reply:
yea, that's the only cause

Reply:
My assumption here is that we're not sure what goes wrong, so surface what happens.